### PR TITLE
Feat/anonreport slash

### PR DESCRIPTION
**Pull request by tanndlin**

### CODE REVIEW

**Review by c650 — CHANGES_REQUESTED** [review]
this diff is huge. did you mess up a rebase?

**Comment by c650** [comment]
also will anonreport still be a dm?

**Comment by tanndlin** [comment]
> this diff is huge. did you mess up a rebase?

Good catch, I need to rebase the base branch with develop

> also will anonreport still be a dm
No I made it ephemeral for all responses when reporting

**Comment by c650** [comment]
> > also will anonreport still be a dm
> 
> No I made it ephemeral for all responses when reporting

if it's not a dm and someone types it wrong, like putting a ! instead of a / , itll get sent to whatever channel, right? minor concern

File: src/app/slash_plugins/anonreport.plugin.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if (subcommand === 'respond') {
+      if (!isModerator) {
+        return interaction.reply({
+          content: 'You must be a moderator to use this command.',
+          ephemeral: true,
+        });
+      }
+
+      const ticket_id = interaction.options.getString('ticket_id');
+      if (!ticket_id) {
```
Comment (c650) [inline src/app/slash_plugins/anonreport.plugin.ts:82]: is this validation necessary, since we spec that "respond" has a `required` ticket_id param?

**Review by c650 — APPROVED** [review]
code lgtm. 

still have the concern about the fundamental change from this being a dm command to it being in a pub channel

File: src/app/slash_plugins/anonreport.plugin.ts (hunk 1 above)
Comment (tanndlin) [reply src/app/slash_plugins/anonreport.plugin.ts:82]: I can assert it non null, for whatever reason it comes as nullable from the API even though its required

**Comment by tanndlin** [comment]
> if it's not a dm and someone types it wrong, like putting a ! instead of a / , itll get sent to whatever channel, right? minor concern

Hmm I'm not 100% sure on the way to go about this. It would be nice to completely move away from DMs since that was always a hassle

if we remove all DMs we dont have to worry about catching when the user has messages from non friends blocked, etc

**Review by c650 — APPROVED** [review]

**Review by joeydotdev — APPROVED** [review]
🚢

**Pull request merged**